clang/lib/CIR/CodeGen/CIRGenCall.cpp

```diff
@@ -141,6 +190,44 @@ arrangeFreeFunctionLikeCall(CIRGenTypes &cgt, CIRGenModule &cgm,
   return cgt.arrangeCIRFunctionInfo(retType, argTypes, required);
 }
 
+/// Arrange a call to a C++ method, passing the given arguments.
+///
+/// passProtoArgs indicates whether `args` has args for the parameters in the
+/// given CXXConstructorDecl.
+const CIRGenFunctionInfo &CIRGenTypes::arrangeCXXConstructorCall(
+    const CallArgList &args, const CXXConstructorDecl *d, CXXCtorType ctorKind,
+    bool passProtoArgs) {
+
+  // FIXME: Kill copy.
```

Comment:
The only idea I have for us to do something like this, would be to have `arrangeCIRFunctionInfo` to have a optional `arg-transform` function.  So it becomes:

`arrangeCIRFunctionInfo(resultType, argTypes, [&](auto *arg) { astContext.getCanonicalParamType(arg.ty);});`

Alternatively, have it take some sort of mutating range, which does the work for us?  

Or maybe build that into CallArgList, and have `arrangeCIRFunctionInfo` take a `CallArgList` and use a `arg.getCanonicalParamType()` on CallArg's object.  WDYT?

Reply:
This is one of those comments that was brought over from classic codegen. It appears six times in both classic codegen and the CIR incubator codegen, always in conjunction with a call to arrangeLLVMFunctionInfo/arrangeCIRFunctionInfo. The problem is that one of the places it occurs (the static version of arrangeLLVMFunctionInfo/arrangeCIRFunctionInfo) takes a list of types from the caller but then appends additional types based on FunctionProtoType::getExtParameterInfos().

I definitely want to leave it alone for now. It may be a solvable problem, but the priority of solving it appears to be very low.

Reply:
Ah... Its kind of frustrating to see these FIXME's last as long as they do, then get propagated into the new lowering like this.  I guess I can live with it (as we have for years), but we gotta stop letting these FIXMEs last as long as they do:/

Reply:
Agreed. We've been following the practice of using "TODO(cir)" when we introduce new placeholders like this in the CIR code so we can distinguish them from TODOs that we copied over from classic codegen, but I expect that they'll have the same problem if we don't do something about it.